src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -752,6 +758,22 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::check_tx_inputs_ring_members_diff(const transaction& tx) const
+  {
+    const uint8_t version = m_blockchain_storage.get_current_hard_fork_version();
+    if (version >= 6)
+    {
+      for(const auto& in: tx.vin)
+      {
+        CHECKED_GET_SPECIFIC_VARIANT(in, const txin_to_key, tokey_in, false);
+        for (size_t n = 1; n < tokey_in.key_offsets.size(); ++n)
+          if (tokey_in.key_offsets[n] == 0)
```

Comment:
I think it is sufficient. key_offsets is unsigned. These are relative offsets. I don't understand which two offsets you are talking about in the second point. I don't understand the point of the copy.assign comment, since this doesn't do copies.